src/Xamarin.Android.Build.Tasks/Tasks/AndroidTask.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using Microsoft.Build.Utilities;
+using Xamarin.Build;
+
+namespace Xamarin.Android.Tasks
+{
+	// We use this task to ensure that no unhandled exceptions
+	// escape our tasks which would cause an MSB4018
+	public abstract class AndroidTask : Task
```

Comment:
I'm torn between wanting each of these in their own file, and them all being in one place to make it easy to change...